### PR TITLE
Updated to support Apple's new Binary Interface and Notification Format

### PR DESCRIPTION
**Pull request by jim0020**
Updated to support Apple's new Binary Interface and Notification Format : https://developer.apple.com/library/mac/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/Chapters/CommunicatingWIthAPS.html#//apple_ref/doc/uid/TP40008194-CH101-SW4

### CODE REVIEW

**Comment by gwynantj** [comment]
@tompesman is there any reason why this PR hasn't been reviewed/merged?

We are implementing silent push notifications at the moment and require the `priority` parameter to be set in order for more control over what makes it to the device.

**Comment by tompesman** [comment]
@gwynantj I'm unable to test this because I've no setup like this running at the moment. If you and @jim0020 agree that this PR is correct than I'll be happy to merge it.

File: lib/push/message_apns.rb [hunk 1]
```diff
@@ -43,11 +45,32 @@ def alert
       MultiJson.load(string_or_json) rescue string_or_json
     end
 
-    # This method conforms to the enhanced binary format.
-    # http://developer.apple.com/library/ios/#documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CommunicatingWIthAPS/CommunicatingWIthAPS.html#//apple_ref/doc/uid/TP40008194-CH101-SW4
+    def priority
+      if ( properties[:alert].nil? && 
+         properties[:badge].nil? && 
+         properties[:sound].nil? && 
+         properties[:expiry].nil? &&
```
Comment (gwynantj) [inline lib/push/message_apns.rb]: @jim0020 we don't need to check for a nil expiry........we just need to make sure that `alert`, `badge` and `sound` is nil. `expiry` is a required parameter on the message and will never be nil.

Can you remove this please?

**Comment by gwynantj** [comment]
@tompesman I've commented on an issue that exists with this PR. Hopefully @jim0020 will be able to amend. 👍

**Pull request closed without merging**